woocommerce/woocommerce
# Add CLI support to run migrations.

### All Submissions:






### Changes proposed in this Pull Request:



Add support to run migration via the command line. Since the backfill is not implemented, only migration from wp_posts and wp_postmeta to COT is currently supported.

Closes #32917 

### How to test the changes in this Pull Request:

1. Check out this branch and run `composer dump-autoload`. 
2. Run command `wp wc cot count_unmigrated`. You should see an output like `There are 22872 orders to be migrated.`.
3. Run the command `wp wc cot migrate` . You should see an output like 
```
This feature is not production ready yet. Make sure you are not running these commands in your production environment.
There are 22872 orders to be migrated.
Order Data Migration  100% [===========================================================================================================] 1:23 / 1:22
Migration completed.
Success: 22872 orders were migrated in 80.901494 seconds
```
4. Run the command `wp wc cot verify_cot_data`. You should an output like:
```
This feature is not production ready yet. Make sure you are not running these commands in your production environment.
There are 22872 orders to be verified.
Order Data Verification  100% [==============================================================================================================================] 1:39 / 1:38
Verification completed.
Success: 22872 orders were verified in 100.426563 seconds
```
5. Change some random order column value in any of the migrated table and run the verification command again using `wp wc cot verify_cot_data`. It should be able to pick up the modified data like:
```
This feature is not production ready yet. Make sure you are not running these commands in your production environment.
There are 22872 orders to be verified.
Order Data Verification  100% [==============================================================================================================================] 1:50 / 1:31
Verification completed.
Error: 22872 orders were verified in 110.967540 seconds. Errors were found Array
(
    [21145] => Array
        (
            [0] => _order_shipping_tax
        )

)
```
### Other information:




### FOR PR REVIEWER ONLY:
